pkg/webhooks/webhooks.go

```diff
@@ -1,21 +1,218 @@
-// Package webhooks exposes functions called from admission webhook handlers.
 package webhooks
 
 import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
 	"github.com/greymatter-io/operator/pkg/cli"
 	"github.com/greymatter-io/operator/pkg/installer"
-
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
 var (
 	logger = ctrl.Log.WithName("webhooks")
 )
 
-func Register(mgr ctrl.Manager, i *installer.Installer, c *cli.CLI, cc client.Client) {
-	mgr.GetWebhookServer().Register("/mutate-mesh", &admission.Webhook{Handler: &meshDefaulter{Installer: i}})
-	mgr.GetWebhookServer().Register("/validate-mesh", &admission.Webhook{Handler: &meshValidator{Installer: i, Client: cc}})
-	mgr.GetWebhookServer().Register("/mutate-workload", &admission.Webhook{Handler: &workloadDefaulter{Installer: i, CLI: c}})
+type Loader struct {
+	client.Client
+	*installer.Installer
+	*cli.CLI
+	getServer      func() *webhook.Server
+	disableCertGen bool
+	caBundle       []byte
+	cert           string
+	key            string
+}
+
+func New(cl client.Client, i *installer.Installer, c *cli.CLI, get func() *webhook.Server, disableCertGen bool) (*Loader, error) {
+	wl := &Loader{Client: cl, Installer: i, CLI: c, getServer: get, disableCertGen: disableCertGen}
+
+	var err error
+	if !wl.disableCertGen {
+		err = wl.loadCerts()
+	}
+
+	return wl, err
```

Comment:
Use a narrower scope for the error
```suggestion
	if !wl.disableCertGen {
		if err := wl.loadCerts(); err != nil {
		    return nil, err
		}
	}

	return wl, nil
```